### PR TITLE
Add alias and directory customisation points

### PR DESCRIPTION
**Pull request by jryans**
These are useful for alternative environments, such as P2P, where alias and directory handling expectations are a bit different.

### CODE REVIEW

File: res/css/views/elements/_ToggleSwitch.scss [hunk 1]
```diff
@@ -24,6 +24,8 @@ limitations under the License.
 
     background-color: $togglesw-off-color;
     opacity: 0.5;
+
+    cursor: unset;
```
Comment (turt2live) [inline res/css/views/elements/_ToggleSwitch.scss:28]: where is the cursor being set from when disabled? It feels like this is the wrong place to be disabling it if it's showing up from somewhere else.

Also, this feels unrelated?

File: src/Rooms.ts [hunk 2]
```diff
@@ -28,7 +29,18 @@ import { MatrixClientPeg } from './MatrixClientPeg';
  * @returns {string} A display alias for the given room
  */
 export function getDisplayAliasForRoom(room: Room): string {
-    return room.getCanonicalAlias() || room.getAltAliases()[0];
+    return getDisplayAliasForAliasSet(
+        room.getCanonicalAlias(), room.getAltAliases(),
+    );
+}
+
+// The various display alias getters all feed through this one path so there's a
```
Comment (turt2live) [inline src/Rooms.ts:37]: because time is a thing in our project:
```suggestion
// The various display alias getters should all feed through this one path so there's a
```

**Review by turt2live — APPROVED** [review]

File: res/css/views/elements/_ToggleSwitch.scss (hunk 1 above)
Comment (jryans) [reply res/css/views/elements/_ToggleSwitch.scss:28]: > where is the cursor being set from when disabled? It feels like this is the wrong place to be disabling it if it's showing up from somewhere else.

Every `ToggleSwitch` is wrapped in `AccessibleButton`, and AB sets `cursor: pointer` as a base. Perhaps a better fix though is to apply the AB disabled class as well when disabled...?

> Also, this feels unrelated?

Yeah, fair enough, just happened to notice it because the room directory publish setting is toggle. I'll remove from here.

File: src/Rooms.ts (hunk 2 above)
Comment (jryans) [reply src/Rooms.ts]: Haha, thanks, added.
